src/event-stream/event-server.ts

```diff
@@ -866,23 +866,22 @@ export async function startEventServer(opts: {
 
   const app = express();
 
-  const handleRawEventRequest = asyncHandler(async req => {
+  const handleRawEventRequest = async (req: express.Request) => {
     await messageHandler.handleRawEventRequest(req.path, req.body, db);
 
     if (logger.level === 'debug') {
       const eventPath = req.path;
-      let payload = JSON.stringify(req.body);
+      let payload = req.body;
       // Skip logging massive event payloads, this _should_ only exclude the genesis block payload which is ~80 MB.
       if (payload.length > 10_000_000) {
         payload = 'payload body too large for logging';
       }
       logger.debug(`${eventPath} ${payload}`, { component: 'stacks-node-event' });
     }
-  });
+  };
 
   app.use(loggerMiddleware);
-
-  app.use(bodyParser.json({ type: 'application/json', limit: '500MB' }));
+  app.use(bodyParser.text({ type: 'application/json', limit: '500MB' }));
```

Comment:
What happens is the size limit is reached here? Should we make this size larger?

Reply:
The request would be rejected with an HTTP 4xx error (which is generally what we want, as opposed to silently fail). We can increase the limit if we expect an event to be larger than 500MB. However, 500MB raw http request body would already use something like 2GB of memory after JSON parsing and postgres client serialization. I don't think our writer environments expect to have much more memory than that (2GB in payload processing + normal operating memory)